compiler/AST/symbol.cpp

```diff
@@ -303,6 +308,82 @@ bool VarSymbol::isParameter() const {
   return hasFlag(FLAG_PARAM) || immediate;
 }
 
+
+std::string VarSymbol::docsDirective() {
+  std::string result;
+  if (fDocsTextOnly) {
+    result = "";
+  } else {
+    // TODO: If this is a type (i.e. this->hasFlag(FLAG_TYPE_VARIABLE)), use
+    //       ".. type:: " as directive. (thomasvandoren, 2015-02-21)
+    if (this->isField) {
+      result = ".. attribute:: ";
+    } else {
+      result = ".. data:: ";
+    }
+  }
+  return this->hasFlag(FLAG_CONFIG) ? result + "config " : result;
+}
+
+
+void VarSymbol::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->hasFlag(FLAG_NO_DOC) || this->hasFlag(FLAG_SUPER_CLASS)) {
+      return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+
+  if (this->hasFlag(FLAG_TYPE_VARIABLE)) {
+    *file << "type ";
+  } else if (this->isConstant()) {
+    *file << "const ";
+  } else if (this->isParameter()) {
+    *file << "param ";
+  } else {
+    *file << "var ";
+  }
+
+  *file << this->name;
+
+  if (this->defPoint->exprType != NULL) {
+    *file << ": ";
+    this->defPoint->exprType->prettyPrint(file);
+  }
+
+  // FIXME: Do we want this for non-class globals/configs?
+  //        (thomasvandoren, 2015-02-21)
```

Comment:
I think the downside to that approach is that it makes the type difficult to use in any contexts other than methods/functions that take that type.  E.g., a user might want to know:
* Can I write an age()?
* Can I pass/use it somewhere that an int is required?
* Can I cast it to int?  
At present, we rely on the int-ness of age for these capabilities.  If it's a complete black box, it limits the usability of the type.

Moreover, in class/record contexts, knowing that a type has a "default value" (let's call it "default definition" to avoid my previous confusion) is crucial for interpreting the type and knowing how to call it.  That is:

record myRange {
  type idxType = int;
  var low, high, stride: idxType;
}

it's key for me to know that:

var r: myRange();

is a legal definition, i.e., that I don't have to type:

var r: myRange(int);

every time I want to create a range (just as default values for var/const/param fields are similarly useful to know when they have default values).

Given both of these concerns, combined with lack of a case in hand where I'd want to squash the definition of the type, I'd prefer to print type definitions in all cases.